dev/com.ibm.ws.microprofile.reactive.messaging_fat/fat/src/com/ibm/ws/microprofile/reactive/messaging/fat/kafka/mtls/single/KafkaMtlsChannelConnectorTest.java

```diff
@@ -0,0 +1,118 @@
+
+    @BeforeClass
+    public static void setup() throws Exception {
+
+        // Channel Properties should take priority over Connector Properties. So provide valid keystore to channels and invalid to Connector.
+        // If invalid keystore is used then the SSL Handshake will fail
+        ConnectorProperties outgoingProperties = simpleOutgoingChannel(null, BasicMessagingBean.CHANNEL_OUT)
+                // Valid Keystore
+                .addProperty(SslConfigs.SSL_KEYSTORE_LOCATION_CONFIG, KafkaUtils.KEYSTORE_FILENAME)
+                .addProperty(SslConfigs.SSL_KEYSTORE_PASSWORD_CONFIG, MtlsTests.kafkaContainer.getKeystorePassword());
+
+        ConnectorProperties incomingProperties = simpleIncomingChannel(null, BasicMessagingBean.CHANNEL_IN, APP_GROUP_ID)
+                // Valid Keystore
+                .addProperty(SslConfigs.SSL_KEYSTORE_LOCATION_CONFIG, KafkaUtils.KEYSTORE_FILENAME)
+                .addProperty(SslConfigs.SSL_KEYSTORE_PASSWORD_CONFIG, MtlsTests.kafkaContainer.getKeystorePassword());
+
+        ConnectorProperties connectorProperties = new ConnectorProperties(ConnectorProperties.Direction.CONNECTOR, "liberty-kafka")
+                .addAll(MtlsTests.testConnectionProperties())
+                // Add Keystore which contains a key not in the Kafka Server truststore, so would be rejected if used
+                .addProperty(SslConfigs.SSL_KEYSTORE_LOCATION_CONFIG, KafkaUtils.KEYSTORE2_FILENAME)
+                .addProperty(SslConfigs.SSL_KEYSTORE_PASSWORD_CONFIG, MtlsTests.kafkaContainer.getKeystorePassword());
+
+        PropertiesAsset appConfig = new PropertiesAsset()
+                .addProperty(KafkaTestClientProvider.CONNECTION_PROPERTIES_KEY, KafkaTestClientProvider.encodeProperties(MtlsTests.connectionProperties()))
+                .include(incomingProperties)
+                .include(outgoingProperties)
+                .include(connectorProperties);
+
+        WebArchive war = ShrinkWrap.create(WebArchive.class, APP_NAME + ".war")
+                .addAsLibraries(kafkaClientLibs())
+                .addAsManifestResource(kafkaPermissions(), "permissions.xml")
+                .addPackage(KafkaMtlsTestServlet.class.getPackage())
+                .addPackage(BasicMessagingBean.class.getPackage())
+                .addPackage(AbstractKafkaTestServlet.class.getPackage())
+                .addPackage(KafkaTestConstants.class.getPackage())
+                .addAsResource(appConfig, "META-INF/microprofile-config.properties");
+
+        ShrinkHelper.exportDropinAppToServer(server, war, SERVER_ONLY);
+
+        KafkaUtils.copyTrustStore(MtlsTests.kafkaContainer, server);
+        KafkaUtils.copyKeyStoresToServer(MtlsTests.kafkaContainer, server);
+
+        server.startServer();
+    }
+
+    @Test
+    public void testMtlsConnector() throws Exception {
+        runTest(server, APP_NAME + "/KafkaMtlsTestServlet", "testMtls");
+    }
```

Comment:
As before, use `@Test` on the servlet method and `@TestServlet` on the `LibertyServer` field.